### DISCUSSION TITLE
The button gets clicked and then fails the test showing a timeout error intermittently

### DISCUSSION THREAD

**Discussion by Neha-test-dev** [getgauge/gauge#1828] [General]
I'm trying to execute my test using Gauge and Taiko. To login into the application, the username and  password are entered in a new window which is opened during the login process. The last step is to click on Yes to complete the login process. The Yes button gets clicked and logins the user successfully. The test fails most of the times with a timeout error in clicking that element("Yes")though the button was clicked. Sometimes with the same code, the tests pass successfully. Is there any issue with clicking buttons? 
As the element to be clicked is already in focus I have tried to do the Press action as well resulting with similar experience as above. Please share your inputs if you have any solution to the problem I'm currently facing.

**Comment by kskarra** [comment]
This is a farily old post, so I guess you've solved this. But kinda looks like your timeout is a little low, i.e. if the page loads quick it works, but if it's slow it times out.

**Comment by sswaroopgupta** [comment]
You can set waitForNavigation:true. There are several wait options you can choose from, here https://docs.taiko.dev/api/tap/. Sometimes it may take more than 3000ms(which is the default) to navigate. You can override it using the navigationTimeout option. Here is an example 
```await click("Login",{waitForNavigation:true,navigationTimeout:5000});```. 
Hope this helps.
